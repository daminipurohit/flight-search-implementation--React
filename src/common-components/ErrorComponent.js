import React from 'react';
import PropTypes from 'prop-types';
import '../styles/ErrorComponent.scss';

const ErrorComponent = ({ errorMsg = 'Something went wrong. Please try again in sometime.' }) => (
    <div className="error-container">
        <div className="error-message">{errorMsg}</div>
    </div>
);

ErrorComponent.propTypes = {
    errorMsg: PropTypes.string
};
export default ErrorComponent;
