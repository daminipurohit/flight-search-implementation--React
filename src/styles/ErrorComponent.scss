
@import "./colors";

.error-container {
    background-color: $white;
    font-size: 12px;
    display: flex;
    justify-content: center;
    align-items: center;
    border-radius: 4px;
    padding: 10px;
}