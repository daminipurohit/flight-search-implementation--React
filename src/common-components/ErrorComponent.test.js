import React from 'react';
import { shallow } from 'enzyme';
import ErrorComponent from './ErrorComponent';

describe('ErrorComponent', () => {
    it('should render correctly', () => {
        const component = shallow(<ErrorComponent />);
        expect(component).toMatchSnapshot();
    });
});
