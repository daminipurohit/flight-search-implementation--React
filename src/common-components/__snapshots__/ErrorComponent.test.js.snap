// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorComponent should render correctly 1`] = `
<div
  className="error-container"
>
  <div
    className="error-message"
  >
    Something went wrong. Please try again in sometime.
  </div>
</div>
`;
